### PR TITLE
[Fix] Ensure that the collated dance scores in the event admin display in thier defined sort order

### LINKED ISSUE #26: [BUG] Collated results do not explicitly have the correct score category order
**Describe the bug**

We ensured that the order or the dance scores on the score categories is correct in the dances by ensuring we cort then at the server before sending them to the client. 

However in testing we identified that collated has the same problem and that the collated scores are in the correct order by chance. 

This is likely due to the way that the compeientation is originally setup by the system and the order it creates the score categories in the db.  Therefore if we want to guarantee that header consistency to support with the client side csv export then we need to ensure that the ordering of the score categories is explicit for the collated data. 

**To Reproduce**

This is tricky to re-produce and I recommend using live data for 2014 (were the issue was witnessed) and possibly the same for 2015 as this is before we had Angular clients and setup the competitions through the API. 

If you do see with that data proceed to those competitions and then to the results. See on the collated that the column headers for the score categories are not in the standard order. 

**Expected behavior**

**Screenshots**
image

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by davidsmonkeys**
## Description

Order the score categories in the collated results to ensure that thy are presented to the front end in the correct sort order by their sort order property.

This is in order that the score categories present in the same order for dances and collated. This allows for cleaner export of the csv and subsequently working with the data once downloaded from the client. 

It is noted that the CSV export is from the client data only and not from the API. 

Fixes #26  (issue)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

At this point these changes have been validated locally and will be checked on staging once the PR  has been accepted. 

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules 
### :bulb: PR Summary generated by FirstMate

**Overview**: Fixed the sorting of collated dance scores in the event admin.

**Changes**:
*Sorting Logic Update:*
- Modified the loop in `ResultsAuthController.cs` to order `ScoreCategories` by `SortOrder`.
- This ensures that dance scores are displayed in the defined sort order.

**TLDR**: The sorting of dance scores in the event admin is now based on the defined `SortOrder`. Focus on the changes in `ResultsAuthController.cs`.

<sub><sup>Generated by FirstMate and automatically updated on every commit.</sup></sub>

### CODE REVIEW

**Review by dertinfo — APPROVED** [review]
That simple eh. Great
